packages/manager/src/features/StackScripts/UserDefinedFieldsPanel/FieldTypes/UserDefinedSelect.tsx

```diff
@@ -34,88 +30,71 @@ interface Props {
-              return (
-                <MenuItem value={choice} key={index}>
-                  {choice}
-                </MenuItem>
-              );
-            })}
-          </TextField>
-        </div>
-      );
-    } else {
-      return (
-        <div className={classes.root}>
-          {error && <Notice error text={error} spacingTop={8} />}
-          <InputLabel className={classes.radioGroupLabel}>
-            {field.label}
-            {!isOptional && '*'}
-          </InputLabel>
+  const handleSelectOneOf = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const selectedValue = e.target.value;
+    updateFormState(field.name, selectedValue);
+  };
 
+  if (oneof.length > 4) {
+    return (
+      <div>
+        {error && <Notice error text={error} spacingTop={8} />}
+        <TextField
+          label={field.label}
+          onChange={handleSelectOneOf}
+          value={value}
+          select
+        >
           {oneof.map((choice: string, index) => {
             return (
-              <React.Fragment key={index}>
-                <FormControlLabel
-                  value={choice}
-                  control={
-                    <Radio
-                      name={choice}
-                      checked={!!value && value === choice}
-                      /*
-                      NOTE: Although the API returns a default value and we're auto selecting
-                      a value for the user, it is not necessary to store this value
-                      in the state because it's not necessary for the POST request, since
-                      the backend will automatically POST with that default value
-                    */
-                      onChange={this.handleSelectOneOf}
-                      data-qa-perm-none-radio
-                    />
-                  }
-                  label={choice}
-                />
-              </React.Fragment>
+              <MenuItem value={choice} key={index}>
+                {choice}
+              </MenuItem>
             );
           })}
-        </div>
-      );
-    }
+        </TextField>
+      </div>
+    );
   }
-}
+  return (
+    <div className={classes.root}>
+      {error && <Notice error text={error} spacingTop={8} />}
+      <InputLabel className={classes.radioGroupLabel}>
+        {field.label}
+        {!isOptional && '*'}
+      </InputLabel>
 
-const styled = withStyles(styles);
+      {oneof.map((choice: string, index) => {
+        return (
+          <React.Fragment key={index}>
+            <FormControlLabel
+              value={choice}
+              control={
+                <Radio
+                  name={choice}
+                  checked={!!value && value === choice}
+                  onChange={handleSelectOneOf}
+                  data-qa-perm-none-radio
+                />
+              }
+              label={choice}
+            />
+          </React.Fragment>
+        );
+      })}
```

Comment:
Can this be cleaned up by removing the react fragment and using an implicit return?

```suggestion
      {oneof.map((choice: string, index) => (
        <FormControlLabel
          value={choice}
          key={index}
          control={
            <Radio
              name={choice}
              checked={!!value && value === choice}
              onChange={handleSelectOneOf}
              data-qa-perm-none-radio
            />
          }
          label={choice}
        />
      ))}
```